internal/fingerprint/fingerprint.go

```diff
@@ -162,29 +162,32 @@ func (f *Fingerprinter) FingerprintFiles(rootPath string, exclusions []string) (
 }
 
 func computeMD5ForFileAndZip(fileInfo os.FileInfo, path string, exclusions []string) ([]FileFingerprint, error) {
-	fingerprints := []FileFingerprint{}
-
 	if !shouldProcessFile(fileInfo, exclusions, path) {
-		return fingerprints, nil
+		return nil, nil
 	}
 
-	// Scan the contents of compressed files
-	// such as .jar and .nupkg
+	var fingerprints []FileFingerprint
+
+	// If the file should be unzipped, try to unzip and fingerprint it
 	if shouldUnzip(path) {
 		fingerprintsZip, err := inMemFingerprintingCompressedContent(path, exclusions)
 		if err != nil {
-			return nil, err
+			if strings.Contains(err.Error(), "not a valid zip file") {
```

Comment:
I'm not a Go professional, but I've seen comparison of the error type instead of looking at the string. I think that's more clean? In this case it would be like this, to compare against the `ErrFormat` type.

```
if errors.Is(err, zip.ErrFormat) {
```

Reply:
Yes, it is. Old habit.